src/DatePickerInput.js

```diff
@@ -215,10 +209,10 @@ const DatePickerInput = React.createClass({
     return (
       <div
         className={cx('react-datepicker-component', className)}
-        style={style}
-        onClick={this.stopPropagation}>
+        style={style}>
         <div className='react-datepicker-input'>
           <input
+            ref='datePickerInput'
```

Comment:
putting the `ref` on the `input` relies on the fact that `DatePicker` is stopping the propagation of click events. I think we should allow that propagation and check if the event is inside the main `div` or in one of its children nodes.

This how I'm testing it recursively in another component of mine https://github.com/buildo/react-components/blob/master/src/scroll/ScrollView.js#L63-L71